apps/dg/views/component_view.js

```diff
@@ -516,10 +535,20 @@ DG.ComponentView = SC.View.extend(
                 var tMinWidth = this.get('contentMinWidth') || kMinSize,
                     tLoc = Math.max(evt.pageX, tMinWidth),
                     tNewWidth = DG.ViewUtilities.roundToGrid(info.width + (tLoc - info.pageX)),
-                    tMaxWidth = this.parentView.get('contentMaxWidth') || kMaxSize;
+                    tMaxWidth = this.parentView.get('contentMaxWidth') || kMaxSize,
+                    tContainerWidth = this.getContainerWidth();
+                if (DG.KEEP_IN_BOUNDS_PREF) {
+                  var tInspectorDimensions = this.parentView.getInspectorDimensions();
+                  tMaxWidth = Math.min(tMaxWidth, tContainerWidth - (info.left + tInspectorDimensions.width));
+                }
                 // Don't let width of component become too small
                 tNewWidth = Math.min(Math.max(tNewWidth, tMinWidth), tMaxWidth);
                 this.parentView.adjust('width', tNewWidth);
+                if (DG.KEEP_IN_BOUNDS_PREF) {
+                  var tLayout = this.parentView.getPath('layout'),
```

Comment:
`getPath` is normally used only when there is a path. Here you could use
`tLayout = this.getPath('parentView.layout')`

`get` is preferred when what's being fetched is a direct property of the object as in `aView.get('layout')`.

Reply:
I was a bit confused about some of the SproutCore usage on these issues.  Thanks for clarifying.  What do we mean by having a "path?"  I'll make relevant changes.

Reply:
Fixed in latest commit.